api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -164,47 +165,92 @@ public void removeMember(Integer memberId) {
 		removeMember(new Patient(memberId));
 	}
 	
+	/**
+	 * @since 2.1.0
+	 */
 	public void removeMember(Patient patient) {
-		List<CohortMembership> memberToRemoveList = getMembers().stream()
-				.filter(m -> m.getPatient().getPatientId().equals(patient.getPatientId())).collect(Collectors.toList());
-		memberToRemoveList.forEach(m -> m.setEndDate(new Date()));
+		CohortMembership memberToRemove = this.getActiveMembership(patient);
+		if (memberToRemove.getPatient().getPatientId().equals(patient.getPatientId())) {
```

Comment:
Good question, if patient is null for methods that add or remove patients or memberships, the code should just return immediately. Same applies to similar methods in CohortService. 

Do we want these methods to have a boolean return type? Where true is returned only if the patient or member is actually added or removed.